hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/fs/CachingSpaceUsageSource.java

```diff
@@ -94,7 +95,16 @@ public void incrementUsedSpace(long usedSpace) {
   }
 
   public void decrementUsedSpace(long reclaimedSpace) {
-    cachedValue.addAndGet(-1 * reclaimedSpace);
+    long current = cachedValue.get();
+    long newValue = current - reclaimedSpace;
+    if (newValue < 0) {
```

Comment:
Since usedSpace value can be decremented till ZERO and beyond that if we try to decrement, it may go negative, so this logic also not solving the problem completely. E.g. lets say we have current used space as 10 and `reclaimedSpace` is 15, ideally this should not happen, but our logic should be more resilient, in this scenario, we still should set the current used space value to ZERO because going negative is not allowed.

Reply:
What my thought was that retaining the previous value can be more stable in terms of integrity, as it avoids making a potentially incorrect assumption about the space usage. If the decrement was incorrect due to a temporary error or miscalculation, keeping the previous value ensures that once the error is corrected, your cached value remains as accurate as possible.

But I do agree with you on setting to zero when an operation would make the value negative is generally safer and simpler where, regardless of input errors, the system maintains a coherent state.

Thanks for the comment @devmadhuu Ill change the code to set the value to zero.